internal/app/cli/attachment.go

```diff
@@ -107,13 +108,31 @@ func AttachmentDelete(ctx context.Context, spec model.RepoSpec, tmNameOrId, atta
 
 	return err
 }
-func AttachmentFetch(ctx context.Context, spec model.RepoSpec, tmNameOrId, attachmentName string, concat bool) error {
+func AttachmentFetch(ctx context.Context, spec model.RepoSpec, tmNameOrId, attachmentName string, concat bool, outputPath string) error {
 	content, err := commands.AttachmentFetch(ctx, spec, toAttachmentContainerRef(tmNameOrId), attachmentName, concat)
 	if err != nil {
 		Stderrf("Failed to fetch attachment %s to %s: %v", attachmentName, tmNameOrId, err)
 	}
 
-	fmt.Print(string(content))
+	if outputPath == "" {
+		fmt.Print(string(content))
+		return nil
+	}
+	f, err := os.Stat(outputPath)
+	if err != nil && !os.IsNotExist(err) {
+		Stderrf("Could not stat output folder: %v", err)
+		return err
+	}
+	if f != nil && !f.IsDir() {
+		Stderrf("output target folder --output is not a folder")
+		return errors.New("output target folder --output is not a folder")
+	}
+	fullFilename := filepath.Join(outputPath, attachmentName)
+	err = os.WriteFile(fullFilename, content, 0660)
+	if err != nil {
+		Stderrf("could not write attachment to file %s: %v", fullFilename, err)
+		return err
+	}
```

Comment:
Somehow the output path of of the `-o` flag is not automatically created if it does not yet exist.  
I get an error e.g. `could not write attachment to file .fetchtest\readme.md: open .fetchtest\readme.md: The system cannot find the path specified.`  (tested on windows)

Note that for fetching a ThingModel the output path is automatically created if it does not yet exist.

Reply:
fixed

Reply:
The output path is now created, also if it does not yet exists.  

But there is another problem:

If I fetch an attachment that does not exists, I get correctly the message e.g.:
`Failed to fetch attachment readme1.md to nexus-x/meteocontrol/bluelog-xc/v1.0.120-20241002095710-2d7451305ad4.tm.json: attachment not found`
But there is an empty file written to the `-o` output path, in the example above an empty `readme1.md` is written.

Reply:
turns out this problem has previously existed, but it used to show itself only in the incorrect rc=0 when attachment cannot be fetched :-)
fixed now